src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -208,12 +208,22 @@ public Bundle transformDocument(ClinicalDocument cda) {
 
         // transform the sections
         for(Section cdaSec: ccd.getSections()) {
-            SectionComponent fhirSec = resTransformer.tSection2Section(cdaSec);
             
-            if(fhirSec == null)
-            	continue;
-            else
-                ccdComposition.addSection(fhirSec);
+        	SectionComponent fhirSec = null; 		
+        	
+        	
+        	//Conditional logic to only return supported sections.
+        	if (cdaSec instanceof AllergiesSection || cdaSec instanceof ImmunizationsSection || cdaSec instanceof ProceduresSection || cdaSec instanceof ProblemSection || cdaSec instanceof MedicationsSection) {    		
```

Comment:
I don't think this is 100% necessary considering the following conditionals check specifically for the classes they're transforming.

Reply:
You actually need both. Just letting all of the secs get transformed means it can get on the output because the cdaSec instances still exist.